src/renderer/components/editorWithTabs/editor.vue

```diff
@@ -902,52 +902,80 @@ export default {
         headerFooterStyled,
         htmlTitle
       } = options
+
       if (!/^pdf|print|styledHtml$/.test(type)) {
         throw new Error(`Invalid type to export: "${type}".`)
       }
 
       const extraCss = getCssForOptions(options)
       switch (type) {
         case 'styledHtml': {
-          const content = await this.editor.exportStyledHTML({
-            title: htmlTitle || '',
-            printOptimization: false,
-            extraCss
-          })
-          this.$store.dispatch('EXPORT', { type, content })
+          try {
+            const content = await this.editor.exportStyledHTML({
+              title: htmlTitle || '',
+              printOptimization: false,
+              extraCss
+            })
+            this.$store.dispatch('EXPORT', { type, content })
+          } catch (err) {
+            log.error(err)
+            notice.notify({
+              title: 'Export HTML',
+              type: 'error',
+              message: `There is something wrong when export ${htmlTitle || 'html'}`
+            })
+          }
           break
         }
         case 'pdf': {
           // NOTE: We need to set page size via Electron.
-          const { pageSize, pageSizeWidth, pageSizeHeight, isLandscape } = options
-          const pageOptions = {
-            pageSize, pageSizeWidth, pageSizeHeight, isLandscape
+          try {
+            const { pageSize, pageSizeWidth, pageSizeHeight, isLandscape } = options
+            const pageOptions = {
+              pageSize, pageSizeWidth, pageSizeHeight, isLandscape
+            }
+
+            const html = await this.editor.exportStyledHTML({
+              title: '',
+              printOptimization: true,
+              extraCss,
+              header,
+              footer,
+              headerFooterStyled
+            })
+            this.printer.renderMarkdown(html, true)
+            this.$store.dispatch('EXPORT', { type, pageOptions })
+          } catch (err) {
+            log.error(err)
```

Comment:
Please redo ` this.printer.renderMarkdown` by calling `clearup` for printing and `PDF`.

Reply:
Maybe it no necessary to call `cleanup`, it will not exec the `this.prointer.renderMarkdown(html, true)` if there are errors happen in `exportStyledHTML`. 

But I still add it, in case of error happen in `this.prointer.renderMarkdown(html, true)`.